repo/packages/birli/package.py

```diff
@@ -21,14 +23,29 @@ def setup_build_environment(self, env):
         build_dir = self.stage.source_path
         env.set('CARGO_HOME', f"{build_dir}/.cargo")
 
+    def get_features(self):
+        features = ["cfitsio-static"]
+        return features
+
     def install(self, spec, prefix):
         cargo = Executable("cargo")
-        cargo("generate-lockfile")
-        cargo("install", "--path=.", f"--root={prefix}", "--features=cfitsio-static")
+        # cargo("generate-lockfile")
+        features = self.get_features()
+        cargo("install", "--path=.", f"--root={prefix}", f"--features={','.join(features)}", "--locked")
 
     @run_after("install")
     @on_package_attributes(run_tests=True)
     def cargo_test(self):
         cargo = Executable("cargo")
-        cargo("test", "--release", "--lib", "--features=cfitsio-static")
+        features = self.get_features()
+        cargo("test", "--release", "--lib", f"--features={','.join(features)}")
+
+# test me on setonix with
```

Comment:
This would be easier to do in spack 0.23? which supports cargo build systems directly I believe. The test data for Birli are self-contained in the repo